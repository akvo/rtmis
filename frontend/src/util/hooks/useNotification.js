import { useMemo } from "react";
import PropTypes from "prop-types";
import { message } from "antd";

function useNotification() {
  return useMemo(() => {
    const notify = ({ type, message: content, ...props }) => {
      message.config({ top: 150 });
      message.open({
        type,
        content,
        style: { textAlign: "right", marginRight: 20 },
        ...props,
      });
    };
    notify.propTypes = {
      type: PropTypes.oneOf(["success", "error", "info", "warning"]),
      message: PropTypes.string,
    };
    return { notify };
  }, []);
}

export default useNotification;
