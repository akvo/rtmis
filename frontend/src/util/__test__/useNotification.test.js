import { useNotification } from "../hooks";
import { render } from "@testing-library/react";
import { useEffect } from "react";
import { message } from "antd";

const TestNotification = ({ notificationBody }) => {
  const { notify } = useNotification();

  useEffect(() => {
    notify(notificationBody);
  }, [notify, notificationBody]);
  return <div>Message: Not Found</div>;
};

jest.mock("antd", () => ({ message: { open: jest.fn(), config: jest.fn() } }));

const notify = {
  content: "Not Found",
  style: {
    marginRight: 20,
    textAlign: "right",
  },
  type: "error",
};

describe("util/hooks", () => {
  test("test Notification that shows error type with default style", () => {
    const notificationBody = {
      type: "error",
      message: "Not Found",
    };
    render(<TestNotification notificationBody={notificationBody} />);
    expect(message.open).toHaveBeenCalledTimes(1);
    expect(message.open).toHaveBeenCalledWith(notify);
    expect(message.open).toMatchSnapshot();
  });
  test("shows Notification when no body is passed", () => {
    const notificationBody = {};
    render(<TestNotification notificationBody={notificationBody} />);
    expect(message.open).toHaveBeenCalledTimes(1);
  });
});
