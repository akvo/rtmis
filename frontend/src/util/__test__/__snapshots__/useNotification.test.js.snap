// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`util/hooks test Notification that shows error type with default style 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "content": "Not Found",
        "style": Object {
          "marginRight": 20,
          "textAlign": "right",
        },
        "type": "error",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
